<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>100</x>
    <y>100</y>
    <width>1400</width>
    <height>900</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Descly</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="mainLayout">

    <!-- QSplitter для разделения панелей -->
    <item>
     <widget class="QSplitter" name="mainSplitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      
      <!-- Левая панель -->
      <widget class="QWidget" name="leftPanel">
       <layout class="QVBoxLayout" name="leftLayout">
        <item>
         <widget class="QTreeView" name="fileTreeView"/>
        </item>
        <item>
         <widget class="QPushButton" name="browseFolderButton">
          <property name="text">
           <string>Browse Folder</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>

      <!-- Центральная панель -->
      <widget class="QWidget" name="centerPanel">
       <layout class="QVBoxLayout" name="centerLayout">
        <item>
         <widget class="QLabel" name="photoPreviewLabel">
          <property name="text">
           <string>No Photo Selected</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
          <property name="scaledContents">
           <bool>false</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding"/>
          </property>
         </widget>
        </item>
        <!-- Добавленный QProgressBar -->
        <item>
         <widget class="QProgressBar" name="progressBar">
          <property name="minimum">
           <number>0</number>
          </property>
          <property name="maximum">
           <number>100</number>
          </property>
          <property name="value">
           <number>0</number>
          </property>
          <property name="textVisible">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>

      <!-- Правая панель -->
      <widget class="QWidget" name="rightPanel">
       <layout class="QVBoxLayout" name="rightLayout">
        <item>
         <widget class="QLineEdit" name="titleField">
          <property name="placeholderText">
           <string>Enter Title</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPlainTextEdit" name="descriptionField">
          <property name="placeholderText">
           <string>Enter Description</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPlainTextEdit" name="keywordsField">
          <property name="placeholderText">
           <string>Enter Keywords</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="rightButtons">
          <item>
           <widget class="QPushButton" name="generateMetadataButton">
            <property name="text">
             <string>Generate Metadata</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="addCategoryButton">
            <property name="text">
             <string>Show Categories</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>

     </widget>
    </item>

    <!-- Нижняя панель -->
    <item>
     <layout class="QHBoxLayout" name="bottomLayout">
      <item>
       <widget class="QPushButton" name="exportCSVButton">
        <property name="text">
         <string>Export CSV</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="uploadForSaleButton">
        <property name="text">
         <string>Upload for Sale</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="settingsButton">
        <property name="text">
         <string>Settings</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>

    <!-- Настройки AI -->
    <item>
     <layout class="QVBoxLayout" name="aiSettingsLayout">
      <item>
       <widget class="QGroupBox" name="settingsGroupBox">
        <property name="title">
         <string>AI Settings</string>
        </property>
        <layout class="QVBoxLayout" name="settingsLayout">
         <item>
          <widget class="QLabel" name="providerLabel">
           <property name="text">
            <string>Select AI Provider:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="aiProviderComboBox">
           <item>
            <property name="text">
             <string>Gemini AI</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>OpenAI</string>
            </property>
           </item>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="apiKeyLabel">
           <property name="text">
            <string>API Key:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="apiKeyLineEdit">
           <property name="placeholderText">
            <string>Enter your API Key</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="saveSettingsButton">
           <property name="text">
            <string>Save Settings</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>

   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar"/>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
</ui>
